### PR TITLE
New Formula supersonic

### PR DESCRIPTION
**Pull request by valkjsaaa**
Supersonic is an ultra-fast, column oriented query engine library written in C++. It provides a set of data transformation primitives which make heavy use of cache-aware algorithms, SIMD instructions and vectorised execution, allowing it to exploit the capabilities and resources of modern, hyper pipelined CPUs. It is designed to work in a single process.

### CODE REVIEW

File: Library/Formula/supersonic.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Supersonic < Formula
+  homepage "https://code.google.com/p/supersonic/"
+  url "https://supersonic.googlecode.com/files/supersonic-0.9.4.tar.gz"
+  sha256 "1592dfd2dc73f0b97298e0d25e51528dc9a94e9e7f4ab525569f63db0442d769"
+
+  depends_on :x11
```
Comment (mistydemeo) [inline Library/Formula/supersonic.rb]: Does this have a GUI that requires X11?

File: Library/Formula/supersonic.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--without-re2"
+    system "make", "clean"
+    system "make", "check"
+    system "make", "install"
+  end
+
+  test do
+    system "false"
```
Comment (mistydemeo) [inline Library/Formula/supersonic.rb]: Can you add a test that tests the application's functionality? The info on what we like to see is [in the formula cookbook](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md#add-a-test-to-the-formula).

File: Library/Formula/supersonic.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  url "https://supersonic.googlecode.com/files/supersonic-0.9.4.tar.gz"
+  sha256 "1592dfd2dc73f0b97298e0d25e51528dc9a94e9e7f4ab525569f63db0442d769"
+
+  depends_on :x11
+  depends_on "glog"
+  depends_on "protobuf"
+  depends_on "pkg-config" => :build
+
+  def install
+    args=[]
```
Comment (mistydemeo) [inline Library/Formula/supersonic.rb]: Looks like this variable is unused - can you remove this line?

**Comment by valkjsaaa** [comment]
Thanks for comment. There is a `make check` in the Makefile script. I wonder how can I use that in test?

**Comment by mistydemeo** [comment]
The purpose of the test block isn't to run compile-time tests, but to test the software's functionality. Could you try to run a test with some test data?

**Comment by valkjsaaa** [comment]
I see. I will try to write a test.

**Comment by tdsmith** [comment]
Does this need `depends_on "boost"`?

**Comment by valkjsaaa** [comment]
Yes. I forget to include it...

**Comment by valkjsaaa** [comment]
I have fixed all the errors mentioned above and added a test.

File: Library/Formula/supersonic.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on "boost"
+  depends_on "pkg-config" => :build
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--without-re2"
+    system "make", "clean"
```
Comment (mikemcquaid) [inline Library/Formula/supersonic.rb]: Why do you need to run `clean` here?

File: Library/Formula/supersonic.rb [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    system "make"
+    system "make", "check"
```
Comment (mikemcquaid) [inline Library/Formula/supersonic.rb]: How long does this take to run?

File: Library/Formula/supersonic.rb [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+          cout << "Results: ";
+          for(int i = 0; i < 8; i++) {
+              cout << result[i] << " ";
+          }
+
+          return 0;
+      }
+    EOS
+    system ENV.cxx, "test.cpp", "-std=c++1y", "-lsupersonic", "-lglog", "-lprotobuf", "-lboost_system", "-o", "test"
+    system "./test"
```
Comment (mikemcquaid) [inline Library/Formula/supersonic.rb]: Nice test :

**Comment by mikemcquaid** [comment]
Please run `brew audit --strict supersonic` and fix the warnings: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/23115/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/audit_supersonic___strict/

Also, it's failing to install on 10.8. Perhaps it needs a `needs :cxx11`?

File: Library/Formula/supersonic.rb (hunk 4 above)
Comment (valkjsaaa) [inline Library/Formula/supersonic.rb]: I thought the answer is positive. I have tried to remove it on my computer and it failed.

Comment (mikemcquaid) [inline Library/Formula/supersonic.rb]: Please add a Ruby comment explaining that, thanks!

**Comment by valkjsaaa** [comment]
I fixed the trailing whitespace. I'm not sure how to fix the c++11 problem. Can you give me a hint?

**Comment by DomT4** [comment]
If it's an ` cxx11 ` issue something like this:

```ruby
depends_on "boost"

needs :cxx11

def install
  ENV.cxx11
```

**Comment by valkjsaaa** [comment]
Thanks for your comment. It helps a lot. But build fail keeps happen on Mountain Lion. I have tried many solutions but nothings works... I thought that this might be a protobuf problem?

**Comment by tdsmith** [comment]
If you need dependencies to use libc++, you probably need to write something like:

```ruby
with_libcxx = MacOS.version < :mavericks ? ["c++11"] : []
depends_on "boost" => with_libcxx
depends_on "protobuf" => with_libcxx
```

@mikemcquaid et al, is that style acceptable?

**Comment by mistydemeo** [comment]
@tdsmith Yep :+1:

**Comment by valkjsaaa** [comment]
Thanks a lot to you all. I have finally eliminated every error on Mountain Lion.

File: Library/Formula/supersonic.rb [hunk 7]
```diff
@@ -0,0 +1,99 @@
+class Supersonic < Formula
+  homepage "https://code.google.com/p/supersonic/"
+  url "https://supersonic.googlecode.com/files/supersonic-0.9.4.tar.gz"
+  sha256 "1592dfd2dc73f0b97298e0d25e51528dc9a94e9e7f4ab525569f63db0442d769"
+
+  with_libcxx = MacOS.version < :mavericks ? ["c++11"] : []
```
Comment (mikemcquaid) [inline Library/Formula/supersonic.rb]: Just use an `if` block below, thanks!

Comment (valkjsaaa) [inline Library/Formula/supersonic.rb]: I thought @mistydemeo said it is OK?

Comment (mikemcquaid) [inline Library/Formula/supersonic.rb]: I'll let her chip in here.

Comment (valkjsaaa) [inline Library/Formula/supersonic.rb]: Anyway, I can fix it...

Comment (valkjsaaa) [inline Library/Formula/supersonic.rb]: Fixed.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for  new formulae is `supersonic 0.9.4 (new formula)`.

**Comment by valkjsaaa** [comment]
OK. Thanks!

**Pull request closed without merging**